test/unit/ui/control/scale.test.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const test = require('mapbox-gl-js-test').test;
+const window = require('../../../../src/util/window');
+const Map = require('../../../../src/ui/map');
+const ScaleControl = require('../../../../src/ui/control/scale_control');
+
+function createMap() {
+    const container = window.document.createElement('div');
+    return new Map({
+        container,
+        style: {
+            version: 8,
+            sources: {},
+            layers: []
+        },
+        hash: true
+    });
+
+}
+
+test('ScaleControl appears in bottom-left by default', (t) => {
+    const map = createMap();
+    map.addControl(new ScaleControl());
+
+    t.equal(map.getContainer().querySelectorAll('.mapboxgl-ctrl-bottom-left .mapboxgl-ctrl-scale').length, 1);
+    t.end();
+});
+
+test('ScaleControl appears in the position specified by the position option', (t) => {
+    const map = createMap();
+    map.addControl(new ScaleControl(), 'top-left');
+
+    t.equal(map.getContainer().querySelectorAll('.mapboxgl-ctrl-top-left .mapboxgl-ctrl-scale').length, 1);
+    t.end();
+});
+
+test('ScaleControl should change unit of distance after calling setUnit', (t) => {
+    const map = createMap();
+    const scale = new ScaleControl();
+    map.addControl(scale);
+
+    t.equal(scale.options.unit, 'metric');
+    scale.setUnit('imperial');
+    t.equal(scale.options.unit, 'imperial');
```

Comment:
The fact that the unit is stored in `scale.options.unit` is an implementation detail that could be changed in a future refactor.  Let's update this test so that it wouldn't be affected by such a change by having these assertions actually check the content of the rendered scale control, e.g. maybe by searching for an appropriate string within `map.getContainer().querySelectorAll('.mapboxgl-ctrl-top-left .mapboxgl-ctrl-scale').innerHTML`